### PR TITLE
[kube-prometheus-stack] Label promstack-operator pods with k8s recommended labels

### PR DESCRIPTION
**Pull request by koooge**
Hi there,

#### What this PR does / why we need it
This PR adds missing k8s recommended labels to promstack-operator pods. The labels are the same as ServiceAccount/promstack-operator.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by jkroepke — APPROVED** [review]
LGTM

File: charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml [hunk 1]
```diff
@@ -27,6 +27,8 @@ spec:
     metadata:
       labels:
         app: {{ template "kube-prometheus-stack.name" . }}-operator
+        app.kubernetes.io/name: {{ template "kube-prometheus-stack.name" . }}-prometheus-operator
+        app.kubernetes.io/component: prometheus-operator
```
Comment (GMartinez-Sisti) [inline charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:31]: Any particular reason for not using `{{ include "kube-prometheus-stack.labels" . | indent x }}` as we do everywhere else?

Comment (jkroepke) [reply charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:31]: You are right, the label are used everywhere, including on dashboard which has nothing todo with prometheus-operator.

**Review by jkroepke — CHANGES_REQUESTED** [review]
Please also update https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/deployment

and

https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job

If we are need the labels on different places, I would recommend create a new templated named `kube-prometheus-stack.prometheus-operator.labels` and attach the k8s recommended labels to ALL resources insides the templates/prometheus-operator directory.

**Review by jkroepke — COMMENTED** [review]
Thanks, it LGTM to me.

If possible, I would like to see a conistant style of `<spaces>{{- include "name" . | nindent X }}` everywhere.

You can go to the **file changes** tab here and apply the suggestions as batch.

**Review by jkroepke — APPROVED** [review]
LGTM 👍

**Pull request merged**